pkg/crc/services/dns/dns.go

```diff
@@ -51,13 +53,27 @@ func setupDnsmasq(serviceConfig services.ServicePostStartConfig) error {
 	// Remove the dnsmasq container if it exists during the VM stop cycle
 	_, _, _ = serviceConfig.SSHRunner.Run("sudo podman rm -f dnsmasq-crc")
 
-	// Start the dnsmasq container
-	dnsServerRunCmd := fmt.Sprintf("sudo podman run  --ip %s --name dnsmasq-crc -v %s:/etc/dnsmasq.conf -p 53:%d/udp --privileged -d %s",
-		dnsContainerIP, dnsConfigFilePathInInstance, dnsServicePort, dnsContainerImage)
-	if _, _, err := serviceConfig.SSHRunner.Run(dnsServerRunCmd); err != nil {
-		return err
+	sd := systemd.NewInstanceSystemdCommander(serviceConfig.SSHRunner)
+	if state, err := sd.Status("dnsmasq-crc"); err != nil || state == states.NotFound {
+		// Create the dnsmasq container
+		dnsServerCreateCmd := fmt.Sprintf("sudo podman create  --ip %s --name dnsmasq-crc -v %s:/etc/dnsmasq.conf -p 53:%d/udp --privileged %s",
+			dnsContainerIP, dnsConfigFilePathInInstance, dnsServicePort, dnsContainerImage)
+		// Create the systemd file
+		if _, _, err := serviceConfig.SSHRunner.Run(dnsServerCreateCmd); err != nil {
+			return err
+		}
+		// Generate the systemd unit file
+		if _, _, err := serviceConfig.SSHRunner.Run("sudo podman generate systemd --restart-policy=no dnsmasq-crc | sudo tee /etc/systemd/system/dnsmasq-crc.service"); err != nil {
+			return err
+		}
+		if err := sd.DaemonReload(); err != nil {
+			return err
+		}
+		if err := sd.Enable("dnsmasq-crc.service"); err != nil {
+			return err
+		}
```

Comment:
Shouldn't this be all done by snc with this commit only removing the code manually starting the container?

Reply:
@cfergeau we can create systemd unit file as part of snc but since we update the config (which need instance IP ) as part of crc start so we might need to restart the service after updating the config file. In the snc side we can have dummy file.

Reply:
Looks like the commit log is only telling half of the story then ;) 
From `man systemd.unit` and https://kinvolk.io/docs/flatcar-container-linux/latest/setup/systemd/environment-variables/ I think we can use a `/etc/systemd/system/dnsmasq-crc.service.d/30-ip.conf` file with the content below. Then we'd only need to overwrite that file for each `crc start` and restart the dnsmasq container rather than recreating the whole systemd unit. 

```
$ cat /etc/systemd/system/dnsmasq-crc.service.d/30-ip.conf

[Service]
Environment=DNSMASQ_LISTEN_IP=...
```

Reply:
@cfergeau it is not only one place where this IP is needed and listening to but this template https://github.com/code-ready/crc/blob/master/pkg/crc/services/dns/template.go#L11-L21 need to be updated.

Reply:
During `crc start`, update `/etc/systemd/system/dnsmasq-crc.service.d/30-ip.conf`, update `/var/srv/dnsmasq.conf`, `systemctl restart crc-dnsmasq-container`, is this a problem?

Reply:
I suspect this could even be done VM-side, with some `ExecPre` in the `crc-dnsmasq-container` unit reading the environment file with the IPs and doing some `sed` in the config file, but probably better to keep code that is known to work for now :)